sogs/routes/users.py

```diff
@@ -0,0 +1,115 @@
+from .. import db, http
+from ..model import room as mroom
+from ..model.user import User
+from ..web import app
+from . import auth
+
+from flask import abort, jsonify, g, Blueprint, request
+
+# User-related routes
+
+
+users = Blueprint('users', __name__)
+
+
+@users.post("/user/<SessionID:sid>/moderator")
+@auth.user_required
+def set_mod(sid):
+
+    user = User(session_id=sid)
+
+    req = request.json
+    room_tokens, global_mod = req.get('rooms'), req.get('global', False)
+
+    if room_tokens and not isinstance(room_tokens, list):
+        app.logger.warning("Invalid request: room_tokens must be a list")
+        abort(http.BAD_REQUEST)
+
+    mod, admin, visible = (
+        None if arg is None else bool(arg)
+        for arg in (req.get('moderator'), req.get('admin'), req.get('visible'))
+    )
+
+    # Filter out any invalid or redundant arguments:
+    if (admin, mod) == (None, None):
+        app.logger.warning(
+            "Invalid moderator request: at least one of admin/moderator must be specified"
+        )
+        abort(http.BAD_REQUEST)
+    elif (admin, mod) == (True, False):
+        app.logger.warning("Invalid moderator call: admin=True, moderator=False is impossible")
```

Comment:
this should be error level

Reply:
Leaving as warning because this is a client-side error, but still wanted to log it in case a server admin is trying to figure out why some (wrong) client call "doesn't work".